micrometer-core/build.gradle

```diff
@@ -82,6 +82,7 @@ dependencies {
     optionalApi 'io.micrometer:context-propagation'
 
     // kotlin
+    optionalApi enforcedPlatform('org.jetbrains.kotlin:kotlin-bom')
```

Comment:
Consistently enforces the kotlin library versions match the plugin version. Without `enforcedPlatform`, Gradle was resolving higher versions than we wanted.